### ISSUE TITLE
Installation with Homebrew fails when having multi-user setup (admin account)

### ISSUE THREAD

**Issue by Loketony** [FusionAuth/fusionauth-issues#967]
**Installation with Homebrew fails when having multiuser setup (admin account)**

Install of FusionAuth on macOS with multi-user setup (a separate admin account) fails with errors:
```
Unrecognized domain-target specifier.
Please refer to `man launchctl` for explanation of the domain-target specifiers.
Usage: launchctl enable <service-target>
Error: Failure while executing; `/bin/launchctl enable gui/501/homebrew.mxcl.fusionauth-app` exited with 64.
```
.

macOS Catalina 10.15.7
FusionAuth 1.20.1

```
su <admin-account>
brew tap fusionauth/homebrew-fusionauth
brew install fusionauth-app
brew services start fusionauth-app
```

**Comment by Loketony** [comment]
I should add that there is a workaround: switch to the admin user's GUI and start FusionAuth; then it's possible to switch back to your ordinary user account.

**Comment by mooreds** [comment]
Hiya @Loketony ,

I am unable to replicate this bug. I did the following:

* added a new standard user to my mac
* logged into that account
* opened up a terminal
* ran `su <my admin account>', and then the other three steps.

I have exactly the same versions (of catalina and fusionauth) but don't see the error message. 

Can you provide a bit more information? Where are you seeing that error message? Are you launching via the GUI or `brew services start`? Do you see any info in the log files?

**Comment by Loketony** [comment]
Hey @mooreds !

The issue occurs when you are logged in as a standard user and trying to launch FusionAuth in the terminal by su to an admin user. Like this:
```
su <adminuser>
brew services start fusionauth-app
```
Results in:
```
Error: Failure while executing; `/bin/launchctl bootstrap gui/501 \
/Users/<adminuser>/Library/LaunchAgents/homebrew.mxcl.fusionauth-app.plist` exited with 37.
```
But, actually, now I tried to start FusionAuth as a root user:
```
sudo brew services start fusionauth-app
```
 – and it worked – although there was some warnings issued:
```
Warning: fusionauth-app: the first ProgramArguments value does not exist:
  sh
Warning: Taking root:admin ownership of some fusionauth-app paths:
  /usr/local/opt/fusionauth-app
  /usr/local/var/homebrew/linked/fusionauth-app
This will require manual removal of these paths using `sudo rm` on
brew upgrade/reinstall/uninstall.
Warning: fusionauth-app must be run as non-root to start at user login!
==> Successfully started `fusionauth-app` (label: homebrew.mxcl.fusionauth-app)
```
I guess I can live with this since I only use FusionAuth in macOS for development.

**Comment by mooreds** [comment]
@Loketony . Hmmm. Looks like my comment was misformatted, I changed it.

Did my steps (in the updated comment) differ from yours? I thought I replicated the steps (but didn't see the problem).

**Comment by robotdan** [comment]
I don't believe you should be running `brew service start fusionauth-app` as `root`. This will mess all of the permissions up. You want to install and run the `fusionauth-app` service in brew as your normal user account.

**Comment by Loketony** [comment]
The problem is in a protected multi-user set up on macOS is that you have an ADMIN user for installations (such as running brew), and a normal STANDARD user – **_without permissions to administer the computer_** – for running applications and the GUI. And whenever more access are needed, such as installing new software, you need to su to the ADMIN user. That's why you can't run brew as a standard user without admin rights, an admin account – or a standard user with the rights to administer the computer – is needed.

I'm sorry that I was unclear; but the issue will occur when using a standard user without the permissions to administer the computer.

**Comment by Loketony** [comment]
And @robotdan ; I'm using `sudo brew` for running other servers in my developer environment without problems; such as nginx and mongo-db. This is the outcome when I run `brew services list`:
```
Name              Status  User Plist
fusionauth-app    started root /Library/LaunchDaemons/homebrew.mxcl.fusionauth-app.plist
mongodb-community started root /Library/LaunchDaemons/homebrew.mxcl.mongodb-community.plist
nginx             started root /Library/LaunchDaemons/homebrew.mxcl.nginx.plist
unbound           started root /Library/LaunchDaemons/homebrew.mxcl.unbound.plist
```

**Comment by robotdan** [comment]
Ok. We may have to investigate further, I don't know that our brew formula supports this configuration. 

You may also just run FusionAuth using kickstart in a user directory w/out brew.

**Comment by desoi** [comment]
Just ran into the same error executing "brew services start postgresql" within a script executed something like:

`sudo --user=username --login  /script/path/postgresql-install.sh`

The "username" account is an admin account. I think the issue is that you can't setup services in /Users/username/Library/LaunchAgents without having the GUI login to that account active. If there is no easy solution, maybe it could be detected to output a more helpful error message.

**Comment by a0s** [comment]
I've got `Error: Failure while executing; `/bin/launchctl enable gui/501/homebrew.mxcl.postgresql@14` exited with 64.` during remote login to macbook. In the same time my user was not logged in into gui.
After that i logged in into gui (with vnc), then `brew service start postgresql@14` start working as usual.
